Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -1396,17 +1399,32 @@ void AODProducerWorkflowDPL::run(ProcessingContext& pc)
   mTableTrID = 0;
   mGIDToTableID.clear();
 
+  // helper map for fast search of a corresponding class mask for a bc
+  std::unordered_map<uint64_t, uint64_t> bcToClassMask;
+  for (auto& ctpDigit : ctpDigits) {
+    uint64_t bc = ctpDigit.intRecord.toLong();
+    uint64_t classMask = ctpDigit.CTPClassMask.to_ulong();
+    bcToClassMask[bc] = classMask;
+  }
+
   // filling BC table
-  // TODO: get real triggerMask
-  uint64_t triggerMask = 1;
+  uint64_t triggerMask;
   for (auto& item : bcsMap) {
     uint64_t bc = item.first;
+    auto bcClassPair = bcToClassMask.find(bc);
+    if (bcClassPair != bcToClassMask.end()) {
+      triggerMask = bcClassPair->second;
+    } else {
+      triggerMask = 0;
+    }
```

Comment:
Don't we need to account CTP IRs in the collectBCs ?

Reply:
Yes, you are right. Thank you, @shahor02, I will fix it